src/nodes/core/TempNode.js

```diff
@@ -1,5 +1,11 @@
 import Node from './Node.js';
 
+/**
+ * TODO: Explain the purpose of this module. Especially why some nodes are derived
```

Comment:
@sunag It would be good to explain why some some nodes derive from `Node` others from `TempNode` instead. What is the purpose of `TempNode?

Reply:
`TempNode` uses cache management to create temporary variables if the node is used more than once to prevent duplicate calculations.